Automattic/woocommerce-payments
# Remove the redundant network requests in `capture_charge`

Fixes #6358

#### Changes proposed in this Pull Request

By removing the redundant network requests, we aim to make capturing an intent faster.

In `capture_charge`:

* Remove the Get Intent API request, which was earlier used to fetch intent metadata from Stripe. This method now receives metadata from the calling function.
* Remove the Update Intent API request, which was earlier used to update an intent's metadata with metadata from both the order and itself. We now send the merged metadata to Stripe in the Capture Intent request.

#### Testing instructions

Prerequisite: Server#3937

Do this first with `trunk` and then with this branch. Compare the response time in both case: it should be less with this branch.

> [!NOTE]
> On @anu-rock's MacBook Pro 16" with M2 Pro: Before: 9 secs, After: 6 secs.

* Create a new order in WooCommerce frontend with "Cash on delivery" as the payment method.

* Create a payment intent in Stripe Dashboard. Go to: Your test account → Payments → Create payment → Manual payment. Select the "capture funds later" option.

* On the intent details page, scroll down to the "Metadata" section. Create a new key `order_id` with value from the order created in the previous step. Create another key `reader_ID` with value "wisepad".

* Make a POST request to `/wp-json/wc/v3/payments/orders/<order_id>/capture_terminal_payment`.

* Verify that the payment succeeds.

* Also verify on the intent details page that the metadata contains `reader_ID` along with order-related key-value pairs.

##### BONUS: Testing with mobile app

Use this guide (PdfdoF-D-p2) to test with the Woo mobile app and verify that:
1. IPP payment goes through successfully.
2. The payment intent's metadata contains data from both order and app (screenshot 👇).

<img width="519" alt="Screenshot 2023-09-27 at 4 14 10 PM" src="https://github.com/Automattic/woocommerce-payments/assets/1288616/3d2d8e88-b73a-4453-9396-5f2857def710">


-------------------


**Post merge**
